Engine/Modules/Notifications/Models/BackendNotification.php

```diff
@@ -0,0 +1,184 @@
+ * @ORM\Entity
+ */
+class BackendNotification extends AbstractModel {
+    /**
+     * @var int
+     * @ORM\Column(name="id", type="integer", nullable=false)
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="IDENTITY")
+     */
+    private $id;
+
+    /**
+     * @var string
+     * @ORM\Column(name="type", type="string", nullable=false)
+     */
+    private $type;
+
+    /**
+     * @var string
+     * @ORM\Column(name="message", type="string", nullable=false)
+     */
+    private $message;
+
+    /**
+     * @var string
+     * @ORM\Column(name="link", type="string", nullable=true)
+     */
+    private $link;
+
+    /**
+     * @var int
+     * @ORM\Column(name="userId", type="integer", nullable=true)
+     * @ManyToOne(targetEntity="\Oforge\Engine\Modules\Auth\Models\BackendUser")
+     */
+    private $userId = null;
+
+    /**
+     * @var bool
+     * @ORM\Column(name="seen", type="boolean")
+     */
+    private $seen = false;
+
+    /**
+     * @var string
+     * @ORM\Column(name="timestamp", type="datetime", nullable=false);
+     */
+    private $timestamp;
+
+    /**
+     * @var int
+     * @ORM\Column(name="group", type="integer", nullable=true)
+     */
+    private $group = null;
+
+    /**
+     * @return int
+     */
+    public function getId() : int {
+        return $this->id;
+    }
+
+    /**
+     * @return string
+     */
+    public function getType() : string {
+        return $this->type;
+    }
+
+    /**
+     * @return string
+     */
+    public function getLink() : string {
+        return $this->link;
+    }
+
+    /**
+     * @return int
+     */
+    public function getUserId() : int {
+        return $this->userId;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isSeen() : bool {
+        return $this->seen;
+    }
+
+    /**
+     * @return string
+     */
+    public function getTimestamp() : string {
+        return $this->timestamp;
+    }
+
+    /**
+     * @return int
+     */
+    public function getGroup() : int {
+        return $this->group;
+    }
+
+    /**
+     * @return string
+     */
+    public function getMessage() : string {
+        return $this->message;
+    }
+
+    /**
+     * @param string $type
+     *
+     * @return BackendNotification
+     */
+    public function setType(string $type) : BackendNotification {
+        $this->type = $type;
+
+        return $this;
+    }
+
+    /**
+     * @param string $link
+     *
+     * @return BackendNotification
+     */
+    public function setLink(string $link) : BackendNotification {
+        $this->link = $link;
+
+        return $this;
+    }
+
+    /**
+     * @param int $userId
+     *
+     * @return BackendNotification
+     */
+    public function setUserId(int $userId) : BackendNotification {
+        $this->userId = $userId;
+
+        return $this;
+    }
+
+    /**
+     * @return BackendNotification
+     */
+    public function setAsSeen() : BackendNotification {
```

Comment:
Please remove this method and add a "normal" setter. Maybe it's possible to mark a notification as unread.